docs/content/integrations/openai.mdx

```diff
@@ -0,0 +1,146 @@
+
+More information is also available the [OpenAI API documentation website](https://platform.openai.com/docs/introduction).
+
+---
+
+## Prerequisites
+
+To get started, you will need to install the `dagster` and `dagster-openai` Python packages:
+
+```bash
+pip install dagster dagster-openai
+```
+
+Note that you will need an OpenAI [API key](https://platform.openai.com/api-keys) to use the resource, which can be generated in your OpenAI account.
+
+---
+
+## Connecting to OpenAI
+
+The first step in using OpenAI with Dagster is to tell Dagster how to connect to an OpenAI Client using an OpenAI [resource](/concepts/resources). This resource contains the credentials needed to interact with OpenAI API.
+
+We will supply our credentials as environment variables by adding them to a `.env` file. For more information on setting environment variables in a production setting, see [Using environment variables and secrets](/guides/dagster/using-environment-variables-and-secrets).
+
+```bash
+# .env
+
+OPENAI_API_KEY=...
+```
+
+Then, we can instruct Dagster to authorize the OpenAI resource using the environment variables:
+
+```python dedent=4
+from dagster_openai import OpenAIResource
+from dagster import EnvVar
+
+# Pull API key from environment variables
+openai = OpenAIResource(
+    api_key=EnvVar("OPENAI_API_KEY"),
+)
+```
+
+---
+
+## Using OpenAI resource with assets
+
+The OpenAI resource can be used in assets in order to interact with the OpenAI API. Note that in this example, we supply our credentials as environment variables directly when instantiating the `Definitions` objects.
+
+```python dedent=4
+from dagster import (
+    AssetExecutionContext,
+    Definitions,
+    EnvVar,
+    asset,
+    define_asset_job,
+)
+from dagster_openai import OpenAIResource
+
+
+@asset(compute_kind="OpenAI")
+def openai_asset(context: AssetExecutionContext, openai: OpenAIResource):
+    with openai.get_client(context) as client:
+        client.chat.completions.create(
+            model="gpt-3.5-turbo", messages=[{"role": "user", "content": "Say this is a test."}]
+        )
+
+
+openai_asset_job = define_asset_job(name="openai_asset_job", selection="openai_asset")
+
+defs = Definitions(
+    assets=[openai_asset],
+    jobs=[openai_asset_job],
+    resources={
+        "openai": OpenAIResource(api_key=EnvVar("OPENAI_API_KEY")),
+    },
+)
+```
+
+After materializing your asset, your OpenAI usage metadata will be available in the Events and Plots tab of your asset.
+TODO: add images.
+
+---
+
+## Using OpenAI resource with ops
+
+The OpenAI resource can also be used in ops in order to interact with the OpenAI API.
+
+Note that, currently, the OpenAI resource can be used in ops to interact with OpenAI API, but cannot log OpenAI Usage metadata.
```

Comment:
```suggestion
The OpenAI resource can also be used in ops. **Note**: Currently, the OpenAI resource cannot log OpenAI usage metadata when used in ops.
```

I think we can tighten this up a bit. We say 'interact with the OpenAI API' a lot, so I think the user probably gets that at this point.